core/src/com/unciv/ui/civilopedia/CivilopediaText.kt

```diff
@@ -101,17 +201,61 @@ class FormattedLine (
 
         val fontSize = when {
             header in headerSizes.indices -> headerSizes[header]
-            else -> defaultSize
+            size == Int.MIN_VALUE -> defaultSize
+            else -> size
         }
+        val labelColor = if(starred) defaultColor else displayColor
+
         val table = Table(CameraStageBaseScreen.skin)
+        var iconCount = 0
+        val iconSize = max(minIconSize, fontSize * 1.5f)
+        if (linkType != LinkType.None && !noLinkImages) {
+            table.add( ImageGetter.getImage(linkImage) ).size(iconSize).padRight(iconPad)
+            iconCount++
+        }
+        if (iconToDisplay.isNotEmpty() && !noLinkImages) {
+            val parts = iconToDisplay.split('/', limit = 2)
+            if (parts.size == 2) {
+                val category = CivilopediaCategories.fromLink(parts[0])
+                if (category != null) {
+                    if (category.getImage != null) {
+                        // That Enum custom property is a nullable reference to a lambda which
+                        // in turn is allowed to return null. Sorry, but without `!!` the code
+                        // won't compile and with we would get the incorrect warning.
+                        @Suppress("UNNECESSARY_NOT_NULL_ASSERTION") 
+                        val image = category.getImage!!(parts[1], iconSize)
+                        if (image != null) {
+                            table.add(image).size(iconSize).padRight(iconPad)
+                            iconCount++
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
In general there is way too much indentation here. My eyes

Reply:
.. are hopefully just fine. Never forget this project is meant to be `fun`, not ruin our organs ![image](https://user-images.githubusercontent.com/63000004/124384711-df263c80-dcd2-11eb-90ca-523b49430ddd.png). (Who am I telling that?)

Maybe a leftover from trying to get that warning gone the peaceful way. Maybe an inline fun with early `return`'s?